pkg/sources/source.go

```diff
@@ -24,29 +28,43 @@ type SourceProcessor struct {
 	VertexInstance *dfv1.VertexInstance
 }
 
-func (u *SourceProcessor) Start(ctx context.Context) error {
+func (sp *SourceProcessor) Start(ctx context.Context) error {
 	log := logging.FromContext(ctx)
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	var writers []isb.BufferWriter
-	switch u.ISBSvcType {
+
+	// watermark variables no-op initialization
+	var fetchWatermark fetch.Fetcher = generic.NewNoOpWMProgressor()
```

Comment:
Sources need to do it twice (source bucket, and the another normal bucket) right? Where is that implemented?

Reply:
https://github.com/numaproj/numaflow/blob/source-watermark-refactor/pkg/watermark/generic/generic.go#L154